src/shared/dissect-image.c

```diff
@@ -3741,17 +3739,23 @@ int mount_image_privately_interactively(
         if (r < 0)
                 return log_error_errno(r, "Failed to relinquish DM and loopback block devices: %m");
 
+        if (ret_directory) {
```

Comment:
This doesn't need to be conditionalized, or you'll need to remove the assert above, plus moving the assignment of `ret_directory` into this block.